File: seearchingAndSorting/QuickSort.js
let arr = [10, 70, 30, 20, 11, 99];

function swap(a, b) {
  let temp = arr[a];
  arr[a] = arr[b];
  arr[b] = temp;
}

function partition(low, high) {
  let pivot = arr[high];
  let i = low - 1;
  for (let j = low; j < high; j++) {
    if (arr[j] < pivot) {
      i++;
      swap(i, j);
    }
  }
  i++;
  swap(i, high);
  return i;
}

function QuickSort(low, high) {
    if(low <high){
 let pivotIndex = partition(low, high);

  QuickSort(low, pivotIndex - 1);
  QuickSort(pivotIndex + 1, high);
    }
 
}

QuickSort(0, arr.length - 1);
console.log(arr);

File: seearchingAndSorting/quickSort.cpp
#include<iostream>
#include<algorithm>
#include<vector>

using namespace std;

int partition(vector<int>&arr , int  start, int end){
    int pivot = arr[end];
    int i = start-1;
    for(int j=start; j<end;j++){
        if(arr[j]<pivot){
            i++;
            swap(arr[i],arr[j]);
        }
    }
    swap(arr[i+1],arr[end]);
    cout<<i+1<<endl;
    return i+1;
}

void quickSort(vector<int>&arr , int  start, int end){
    if(start<end){
        int pivotIndex = partition(arr, start, end);
    quickSort(arr,start,pivotIndex-1);
    quickSort(arr,pivotIndex+1, end);
    }
   return;
}

int main(){
    vector<int>arr = {10,3,7,11,45,70,5 };
    quickSort(arr, 0, arr.size()-1);

    for(int val:arr){
        cout<<val <<" ";
    }
}